hooks/useDomainNames.tsx

```diff
@@ -1,43 +1,46 @@
-import { useEffect, useState } from 'react'
+import { useState } from 'react'
 
 import {
   getAllDomains,
   performReverseLookupBatch,
 } from '@bonfida/spl-name-service'
+import { AssetAccount } from '@utils/uiTypes/assets'
 
-interface Domains {
+export interface DomainObj {
   domainName: string | undefined
   domainAddress: string
 }
 
-const useDomainsForAccount = (connection, governedAccount) => {
-  const [accountDomains, setAccountDomains] = useState<Domains[]>([])
+const useDomainsForAccount = (connection) => {
+  const [accountDomains, setAccountDomains] = useState<DomainObj[]>([])
   const [isLoading, setIsLoading] = useState(false)
 
-  useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-extra-semi
-    ;(async () => {
-      setIsLoading(true)
-      const domains = await getAllDomains(connection, governedAccount.pubkey)
+  const refreshDomainsForAccount = async (account: AssetAccount) => {
```

Comment:
instead of imperatively fetching the data with an effect, let's make a `react-query` hook that inputs the PublicKey and refreshes automatically and intelligently whenever that input changes. you can look in the `hooks/queries/` folder for examples.